resources/design-document.md

```diff
@@ -0,0 +1,244 @@
+
+U9. As a BTM customer, I want to be able to view all employees on a specific team
+
+U10. As a BTM customer, I want to be able to view all tests taken by an employee
+
+U11. As a BTM customer, I want to be able to deactivate employees that have left
+
+
+Stretch goals
+
+S1. As a BTM customer, I want to have my trainings auto-update their status based off of their date and months to expiration
+
+S2. As a BTM customer, I want to be able to make the next training in a training series by copying the information from the most recent training
+
+
+## 3. Project Scope
+
+### 3.1. In Scope
+
+* Creating, retrieving, and updating an employee
+* Test list that can be searched by training or employee name
+* Recipes can be retrieved and individual recipes added to meal plan
+* A user's meal plan which consists of chosen recipes
+
+### 3.2. Out of Scope
+
+* Trainings auto-updating their status
+* Training Series duplicator for next in series
+
+# 4. Proposed Architecture Overview
+
+This initial iteration will provide the minimum lovable product (MLP) including
+creating, retrieving, and updating an employee, as well as retrieving a training and creating testing for it; and creating a Training Series.
+
+We will use API Gateway and Lambda to create 15(20) endpoints (defined in section 6.2 "Endpoints")
+that will handle the creation, update, and retrieval of Employees, Trainings, Tests and Training Series to satisfy our
+requirements.
+
+We will store employees, trainings, tests, and training series in their own tables in DynamoDB.
+
+Brecghan's Training Matrix(BTM) will also provide a web interface for users to manage
+the matrix. A main page providing connections to view trainings, employees, or test results. From those pages Users
+will be able to create new items and link off to the other viewing pages.
+
+# 5. API
+
+## 6.1. Public Models
+
+```
+// EmployeeModel
+
+String employeeId;
+String employeeName;
+Boolean isActive;
+Enum team;
+Date startDate;
+Set<String> trainingsTaken;
+Set<String> testsTaken;
+Enum trainingStatus;
+```
+
+```
+// TrainingModel
+
+String trainingId;
+String trainingName;
+ZoneDateTime trainingDate;
+Boolean isActive;
+Integer monthsTilExpire;
+Enum expirationStatus;
+Set<String> testsForTraining;
+Set<String> employeesTrained;
+String trainingSeries;
+
+```
+
+```
+// TestModel
+
+String trainingId;
+String employeeId;
+Integer scoreToPass;
+Integer latestScore;
+Boolean hasPassed;
+Integer timesTaken;
+```
+
+```
+// TrainingSeriesModel
+
+String trainingSeriesName;
+```
+
+## 6.2. Endpoints
+
+### Employees
+
+* Accepts `POST` requests to `/employee`
+* Accepts an employeeId, employeeName, start Date, and team; returns the corresponding EmployeeModel including a true boolean parameter isActive, a training status of Up-To-Date, and 2 blank sets of trainingsTaken and testsTaken.
+
+* Accepts `GET` requests to `/employee`
+* Returns a list of all employees.
+
+* Accepts `GET` requests to `/employee?isActive={isActive}`
```

Comment:
Instead of defining this and the team filter as separate endpoints, I would include a note in the main `/employee` endpoint that says something like "Accepts a paramater isActive that will only return active employees" and similar for the team. It's one endpoint, but you can provide optional parameters to filter what's returned.